lib/matplotlib/widgets.py

```diff
@@ -2916,11 +2894,7 @@ class RectangleSelector(_SelectorWidget):
     See also: :doc:`/gallery/widgets/rectangle_selector`
     """
 
-    @_api.rename_parameter("3.5", "maxdist", "grab_range")
-    @_api.rename_parameter("3.5", "marker_props", "handle_props")
-    @_api.rename_parameter("3.5", "rectprops", "props")
-    @_api.delete_parameter("3.5", "drawtype")
-    @_api.delete_parameter("3.5", "lineprops")
+    @_api.make_keyword_only("3.7", name="drawtype")
     def __init__(self, ax, onselect, drawtype='box',
```

Comment:
Everything after `onselect` is already keyword-only because of the `delete_parameter` on `drawtype`, so it should be directly done:
```suggestion
    def __init__(self, ax, onselect, *,
```